File: algorithms/1.two-sum.py
#
# @lc app=leetcode id=1 lang=python3
#
# [1] Two Sum
#

# @lc code=start
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        m = {}
        for i in range(len(nums)):
            m[nums[i]] = i
        for i in range(len(nums)):
            b = m.get(target-nums[i])
            if b and b != i:
                return [i, b]
        return []
# @lc code=end

File: algorithms/1.two-sum.go
package algorithms

/*
 * @lc app=leetcode id=1 lang=golang
 *
 * [1] Two Sum
 */

// @lc code=start
func twoSum(nums []int, target int) []int {
	m := make(map[int]int)
	for i, v := range nums {
		if j, found := m[target-v]; found {
			return []int{i, j}
		}
		m[v] = i
	}
	return []int{-1, -1}
}

// @lc code=end
